instrumentation/resque/lib/opentelemetry/instrumentation/resque/patches/resque_job.rb

```diff
@@ -28,17 +28,18 @@ def perform # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
 
             extracted_context = OpenTelemetry.propagation.extract(@payload)
 
-            links = []
-            if config[:propagation_style] == :link
-              span_context = OpenTelemetry::Trace.current_span(extracted_context).context
-              links << OpenTelemetry::Trace::Link.new(span_context) if span_context.valid?
-            end
-
-            parent_context = (config[:propagation_style] == :child ? extracted_context : OpenTelemetry::Context.current)
-
-            OpenTelemetry::Context.with_current(parent_context) do
-              tracer.in_span(span_name, attributes: attributes, links: links, kind: :consumer) do
-                super
+            OpenTelemetry::Context.with_current(extracted_context) do
+              if config[:propagation_style] == :link
+                links = []
+                span_context = OpenTelemetry::Trace.current_span(extracted_context).context
+                links << OpenTelemetry::Trace::Link.new(span_context) if span_context.valid?
+                OpenTelemetry::Trace.with_span(tracer.start_root_span(span_name, attributes: attributes, links: links, kind: :consumer)) do
+                  super
+                end
```

Comment:
Do you need to call `.finish` on the root span that was created? None of my tests pass unless I do that; I'm wondering if this leads to a dangling span here.

Reply:
🤦 yes, we do. Good catch!

Reply:
Oh just kidding, I see where that's happening later on in another commit. Whoops.